# Домашнее задание № 5

## по дисциплине: Теория вероятностей и математическая статистика

## Тема: Проверка статистических гипотез. P-значения. Доверительные интервалы

### выполнил: Евгений Кириллов

__Задача 1__

Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти ширину доверительного интервала для оценки математического ожидания `mu` с надёжностью 0.95, если выборочное среднее равно 80, а объём выборки равен 256.

__Ответ__: 3.91992797

In [1]:
import numpy as np
from scipy import stats

In [2]:
n = 256
samples_mean = 80
std = 16
p = 0.95
alpha = 1 - p

найдём границы критической области:

In [3]:
t1 = stats.norm.ppf(alpha / 2)
t2 = stats.norm.ppf(1 - alpha / 2)

t1, t2

(-1.959963984540054, 1.959963984540054)

найдём границы доверительного интервала:

In [4]:
(samples_mean + t1 * std / np.sqrt(n), samples_mean + t2 * std / np.sqrt(n))

(78.04003601545995, 81.95996398454005)

посчитаем ширину доверительного интервала:

In [5]:
(samples_mean + t2 * std / np.sqrt(n)) - (samples_mean + t1 * std / np.sqrt(n))

3.9199279690801063

__Задача 2__

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
```
202, 203, 199, 197, 195, 201, 200, 204, 194, 190
```
Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%? Найдите P-значение для данного теста и укажите его в качестве ответа.

__Ответ__: 0.3145666

In [6]:
samples = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
samples

array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

In [7]:
n = samples.shape[0]
n

10

In [8]:
alpha = 0.01

находим значение статистики:

In [9]:
def statistic(samples: np.ndarray) -> float:
    return (samples.mean() - 200) / (samples.std(ddof=1) / np.sqrt(samples.shape[0]))

In [10]:
S = statistic(samples)
S

-1.0651074037450896

находим P-значение:

In [11]:
p_left = stats.t.cdf(S, df=n - 1)
p_right = 1 - stats.t.cdf(S, df=n - 1)

pvalue = 2 * min(p_left, p_right)

pvalue

0.31456660352472854

#### или:

In [12]:
stats.ttest_1samp(samples, popmean=200)

Ttest_1sampResult(statistic=-1.0651074037450896, pvalue=0.31456660352472854)

__Задача 3__

Дана выборка диаметров подшипников из примера 1 с занятия 5:
```
samples = [0.6603, 0.9466, 0.5968, 1.3792, 1.5481, 0.7515, 1.0681, 1.1134,
           1.2088, 1.701 , 1.0282, 1.3579, 1.0191, 1.1784, 1.1168, 1.1372,
           0.7273, 1.3958, 0.8665, 1.5112, 1.161 , 1.0232, 1.0865, 1.02  ]
```
Предполагая, что диаметры подшипников распределены нормально, проверьте гипотезу о том, что дисперсия случайной величины равна `0.0625` при уровне значимости `alpha = 0.05`. Что для этого нужно знать:
1. Альтернативная гипотеза двухсторонняя.
2. Статистика для теста: `H = (n - 1) * sample_variance / variance`, где `n` - число элементов в выборке, `sample_variance` - несмещённая оценка дисперсии, `variance` - утверждаемая нулевой гипотезой дисперсия.
3. Эта статистика в предположении верности нулевой гипотезы имеет распределение хи-квадрат с параметром `df = n - 1`. Её квантили можно найти с помощью функции `scipy.stats.chi2.ppf`.

В качестве ответа запишите 1, если нулевая гипотеза верна, и 0 иначе.

__Ответ__: 1


In [13]:
samples = np.array([0.6603, 0.9466, 0.5968, 1.3792, 1.5481, 0.7515, 1.0681, 1.1134,
                    1.2088, 1.701 , 1.0282, 1.3579, 1.0191, 1.1784, 1.1168, 1.1372,
                    0.7273, 1.3958, 0.8665, 1.5112, 1.161 , 1.0232, 1.0865, 1.02])
samples

array([0.6603, 0.9466, 0.5968, 1.3792, 1.5481, 0.7515, 1.0681, 1.1134,
       1.2088, 1.701 , 1.0282, 1.3579, 1.0191, 1.1784, 1.1168, 1.1372,
       0.7273, 1.3958, 0.8665, 1.5112, 1.161 , 1.0232, 1.0865, 1.02  ])

In [14]:
n = samples.shape[0]
n

24

In [15]:
variance = 0.0625
alpha = 0.05

находим границы критической области:

In [16]:
t1 = stats.chi2.ppf(alpha / 2, df = n - 1)
t2 = stats.chi2.ppf(1 - alpha / 2, df = n - 1)

t1, t2

(11.688551922452438, 38.0756272503558)

находим sample_variance:

In [17]:
sample_variance = samples.std(ddof=1) ** 2
sample_variance

0.07804495041666669

находим значение статистики:

In [18]:
H = (n - 1) * sample_variance / variance
H

28.720541753333343

т.к. 11.688551922452438 < 28.720541753333343 < 38.0756272503558

или, другими словами, значение статистики не попало в критическую область, 

то значит, нулевая гипотеза верна (ответ: 1)